doc/source/user/basics.copies.rst

```diff
@@ -0,0 +1,144 @@
+.. _basics.copies-and-views:
+
+****************
+Copies and views
+****************
+
+When operating on a NumPy array, we can sometimes improve performance by using
+:term:`views <view>` instead of copies of the array. However, when handling
+large amounts of important data, it can be risky to directly operate on it and
+it could be safer to use copies. Hence, it is important to know the difference
+between these two terms to correctly determine when to use which one.
+
+The NumPy array is a :term:`contiguous` block of memory consisting of two parts:
+the data buffer with the actual data elements, and the metadata which contains
+information about the data buffer. The metadata includes data type, strides
+and other important information that helps manipulate the :class:`.ndarray`
+easily. See the :ref:`numpy-internals` section for a detailed look.
+
+View or shallow copy
+====================
+
+It is possible to access the array differently by just changing the
+metadata and without changing the data buffer. This creates a new way of
+looking at the data and these new arrays are called views or shallow copies.
+The data buffer remains the same so any changes made to a view reflects in the
+original copy. A view can be forced through the :meth:`.ndarray.view` method. 
+
+
+Copy or deep copy
+=================
+
+When a new array is created by duplicating the data buffer as well as the
+metadata, it is called a copy or a deep copy. Changes made to the copy
+do not reflect on the original array. Making a copy is slower and memory
+consuming but sometimes necessary. A copy can be forced by using
+:meth:`.ndarray.copy`.
+
+
+How to tell if the array is a view or a copy
+============================================
+
+The :attr:`base <.ndarray.base>` attribute of the ndarray makes it easy
+to tell if an array is a view or a copy. The base attribute of a view returns
+the original array while for a copy it returns ``None``. 
+
+    >>> x = np.arange(9)
+    >>> x
+    array([0, 1, 2, 3, 4, 5, 6, 7, 8])
+    >>> y = x.reshape(3, 3)
+    >>> y
+    array([[0, 1, 2],
+           [3, 4, 5],
+           [6, 7, 8]])
+    >>> y.base  # .reshape() creates a view
+    array([0, 1, 2, 3, 4, 5, 6, 7, 8])
+    >>> z = y[[2, 1]]
+    >>> z
+    array([[6, 7, 8],
+           [3, 4, 5]])
+    >>> z.base is None  # advanced indexing creates a copy
+    True
```

Comment:
It doesn't have to be for this PR, but I think it would be very useful to have a section above this that highlights what sort of operations create copies/views.

The example here has two nice demonstrations of this: reshaping and advanced indexing creating views/copies, respectively. IMO it would be beneficial to highlight these and explain why - in fact I think this should be the main emphasis of this document!

For example, I would move the indexing section and reshaping sections above this one.

This is mostly subjective though - these organizational ideas aren't blockers and we could handle them in followup PRs